pyodide-build/pyodide_build/_py_compile.py

```diff
@@ -0,0 +1,169 @@
+import py_compile
+import shutil
+import sys
+import zipfile
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from packaging.tags import Tag
+from packaging.utils import parse_wheel_filename
+
+
+def _specialize_convert_tags(tags: set[Tag] | frozenset[Tag], wheel_name: str) -> Tag:
+    """Convert a sequence of wheel tags to a single tag corresponding
+    to the current interpreter and compatible with the py -> pyc compilation.
+
+    Having more than one output tag is not supported.
+
+    Examples
+    --------
+    >>> from packaging.tags import parse_tag
+    >>> tags = parse_tag("py2.py3-none-any")
+    >>> str(_specialize_convert_tags(set(tags), ""))
+    'cp310-none-any'
+    >>> tags = parse_tag("cp310-cp310-emscripten_3_1_24_wasm32")
+    >>> str(_specialize_convert_tags(set(tags), ""))
+    'cp310-cp310-emscripten_3_1_24_wasm32'
+    >>> tags = parse_tag("py310.py311-any-none")
+    >>> str(_specialize_convert_tags(set(tags), ""))
+    'cp310-any-none'
+    >>> tags = parse_tag("py36-abi3-none")
+    >>> str(_specialize_convert_tags(set(tags), ""))
+    'cp310-abi3-none'
+    """
+    if len(tags) == 0:
+        raise ValueError("Failed to parse tags from the wheel file name: {wheel_name}!")
+
+    output_tags = set()
+    interpreter = "cp" + "".join(str(el) for el in sys.version_info[:2])
+    for tag in list(tags):
```

Comment:
```suggestion
    for tag in tags:
```

I guess it is okay to iterate from sets directly?